### ISSUE TITLE
i3status battery won't parse Apple Magic Trackpad battery uevent

### ISSUE THREAD

**Issue by edio** [i3/i3status#214]
@Output of `i3 --moreversion 2>&- || i3 --version`:
```
Binary i3 version: 4.13 (2016-11-08) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.13 (2016-11-08) (pid 1994)o abort…)
Loaded i3 config: /home/edio/.config/i3/config (Last modified: Fri 24 Feb 2017 09:56:52 AM CET, 2251835 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

### What I did:

Added to `.config/i3status/config`

```
battery 1 {
    format = "Touchpad: %percentage"
    format_down = "No touchpad"
    path = "/sys/class/power_supply/hid-70:cd:60:f7:a0:5c-battery/uevent"
    low_threshold = 10
}
```

This should have displayed battery status of my Apple Magic Trackpad

```
$ cat /sys/class/power_supply/hid-70:cd:60:f7:a0:5c-battery/uevent
POWER_SUPPLY_NAME=hid-70:cd:60:f7:a0:5c-battery
POWER_SUPPLY_PRESENT=1
POWER_SUPPLY_ONLINE=1
POWER_SUPPLY_CAPACITY=100
POWER_SUPPLY_MODEL_NAME=applemagictrackpad
POWER_SUPPLY_STATUS=Discharging
POWER_SUPPLY_SCOPE=Device
```

### What I saw:

i3status displayed "No touchpad"

### What I expected instead:

i3 status should have displayed actual battery status of my touchpad

**Comment by eplanet** [comment]
Thanks for reporting the issue. Could you provide your i3status version and strace execution as well?
```bash
i3status -v
strace -o /tmp/i3status.log i3status --run-once
```

I see two issues here: accessing the battery file and parsing its content.

For accessing the file, that obviously seems to exist on your computer, there's no reason why the reading function used in [src/general.c#L24](https://github.com/i3/i3status/blob/master/src/general.c#L24) would fail, since `cat` would do exactly the same. The `strace` output should greatly help understand what's going on.

For parsing the file, there's no way that i3status would yield a readable output with the content of your file since it expects either [POWER_SUPPLY_ENERGY_NOW](https://github.com/i3/i3status/blob/master/src/print_battery_info.c#L138) or [POWER_SUPPLY_CHARGE_NOW](https://github.com/i3/i3status/blob/master/src/print_battery_info.c#L141) to be set for current battery. This is the output I get with your file: `Touchpad: -nan%`.

Further reading the [specifications](https://www.kernel.org/doc/Documentation/power/power_supply_class.txt), I understand `POWER_SUPPLY_CAPACITY` could be used as a battery percentage. Maybe @stapelberg @Airblader would confirm we could use it when other values aren't available?

**Comment by stapelberg** [comment]
Sounds good.

**Comment by edio** [comment]
```
$ i3status -v
i3status 2.11 (2017-01-21) © 2008 Michael Stapelberg and contributors
```

strace output is on [pastebin.com/7LwXsZRB](https://pastebin.com/7LwXsZRB)

**Comment by eplanet** [comment]
This issue might be closed since #216 has been merged.
